File: frontend/src/types/design.ts
export type DesignDto = {
  id: number;
  professionalId: number;
  professionalName?: string | null;
  imageUrl: string;
  active: boolean;
  title?: string | null;
};

export type DesignInDto = {
  professionalId: number;
  imageUrl: string;
  title?: string | null;
  active?: boolean;
};

File: backend/src/main/java/com/svalero/RosasTattoo/domain/Design.java
package com.svalero.RosasTattoo.domain;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity(name = "design")
public class Design {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column
    private String title;

    @Column(name = "image_url")
    @NotBlank(message = "An image must be uploaded")
    private String imageUrl;

    @Column(nullable = false)
    private boolean active = true;

    @ManyToOne
    @JoinColumn(name = "professional_id")
    @NotNull(message = "Professional is mandatory")
    private Professional professional;
}
